samples/NativeAOT/NativeAotValueManager.cs

```diff
@@ -1,10 +1,357 @@
+// Originally from: https://github.com/dotnet/java-interop/blob/9b1d8781e8e322849d05efac32119c913b21c192/src/Java.Runtime.Environment/Java.Interop/ManagedValueManager.cs
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
 using Android.Runtime;
+using Java.Interop;
 
 namespace NativeAOT;
 
-// TODO: make a NativeAOT-specific implementation of AndroidValueManager
-// This is enough for "Hello World" to launch
-internal class NativeAotValueManager : AndroidValueManager
+internal class NativeAotValueManager : JniRuntime.JniValueManager
 {
+	readonly NativeAotTypeManager TypeManager;
+	Dictionary<int, List<IJavaPeerable>>?   RegisteredInstances = new Dictionary<int, List<IJavaPeerable>>();
 
+	public NativeAotValueManager(NativeAotTypeManager typeManager) =>
```

Comment:
If this *doesn't* need to know about `NativeAotTypeManager`, then this can instead provide a default constructor and override `OnSetRuntime()` to cache the `TypeManager` instance:

```csharp
public override void OnSetRuntime (JniRuntime runtime)
{
    base.OnSetRuntime (runtime);
    TypeManager = runtime.TypeManager;
}
```

The "benefit" is that you don't need to change the the init code in `JavaInteropRuntime.init()`.  (That might be a silly reason…)

Reply:
It doesn't look like this works because of ordering:

* https://github.com/dotnet/java-interop/blob/fb642c941ebbfd49ba8177acb0bac8d51d75c3d2/src/Java.Interop/Java.Interop/JniRuntime.cs#L184-L188

`runtime.TypeManager` is null at this point.

I changed the field to `JniRuntime.JniTypeManager`, though.